git_theta/lsh/pool.py

```diff
@@ -0,0 +1,64 @@
+"""Class for deterministically supplying pre-computed random values"""
+
+import numpy as np
+import numba as nb
+from numba import int32, float32
+import sys
+
+# TODO(bdlester): importlib.resources doesn't have the `.files` API until python
+# version `3.9` so use the backport even if using a python version that has
+# `importlib.resources`.
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+
+# RandomnessPool is a singleton class so we don't read pool data off disk multiple times per git-theta command
+SINGLETON = None
+
+spec = [("pool", nb.float64[:]), ("index_hashes", nb.int64[:])]
+
+
+@nb.experimental.jitclass(spec)
+class RandomnessPool:
+    def __init__(self, pool, index_hashes):
+        self.pool = pool
+        self.index_hashes = index_hashes
+
+    def get_hyperplanes(self, feature_size):
+        hyperplanes = np.empty((feature_size, self.index_hashes.size))
+        for feature_idx in nb.prange(feature_size):
+            for signature_idx in nb.prange(self.index_hashes.size):
+                hyperplanes[feature_idx, signature_idx] = self.get_hyperplane_element(
+                    feature_idx, signature_idx
+                )
+        return hyperplanes
+
+    def get_hyperplane_element(self, feature_idx, signature_idx):
+        index_hash = self.index_hashes[signature_idx]
+        pool_idx = int(np.mod(np.bitwise_xor(feature_idx, index_hash), self.pool.size))
```

Comment:
Needing this int was suspicious, I was playing around and found I could get different answers when my numbers were `np.uint64`s and I did a bitwise_xor followed by mod with an array as the second argument vs a scalar. When it is a vector I get the right answer (checked by hand) and the right type. When it is a scalar I get the wrong answer and a float return type.

I think numba defaults to unsigned ints for loop variables (I remember reading this but couldn't find it in the docs rn) and when that starts to mix with a scalar that somehow converts everything into floats and we get a floating mod instead of a int mod. The float mod seems to have more collisions in my spot checks too.

I think adding `dtype=np.uint64` (or a normal int64) fixes this in my quick checks.

Reply:
Yeah I was seeing something similar when mod-ing with a scalar. It works if the first argument is `np.int64` but when it's `np.uint64` it returns a float. Just changed the arrays being fed into `bitwise_or` into `np.int64` arrays and it removed the issue.